source/extensions/filters/network/mysql_proxy/route.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/extensions/filters/network/mysql_proxy/v3/mysql_proxy.pb.h"
+#include "envoy/tcp/conn_pool.h"
+#include "envoy/upstream/cluster_manager.h"
+#include "envoy/upstream/thread_local_cluster.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace MySQLProxy {
+
+/**
+ * MySQL route of cluster connection pool by database name.
+ */
+class Route {
```

Comment:
This class and Router are pretty generic, not mysql specific. I am wondering if other filters use similar concepts and code could be reused.